numba/typed/typedlist.py

```diff
@@ -200,12 +200,15 @@ class List(MutableSequence, pt.Generic[T]):
     _legal_kwargs = ["lsttype", "meminfo", "allocated"]
 
     def __new__(cls,
+                *args,
                 lsttype=None,
                 meminfo=None,
                 allocated=DEFAULT_ALLOCATED,
                 **kwargs):
         if config.DISABLE_JIT:
-            return list.__new__(list)
+            pylist = list.__new__(list)
+            pylist.__init__(*args, **kwargs)
+            return pylist
```

Comment:
Wouldn't just `return list(*args, **kwargs)` be less fragile here?

Reply:
Possibly? I felt that the idiom I used was a bit more natural in the context of being in a `__new__()` method... Is there something that could go wrong with this that `list(*args, **kwargs)` would prevent?

Reply:
Yes, theres definitely something that could go wrong. Normally when you call `list(x)`, python calls `l = list.__new__(list, x)` and then `l.__init__(x)`. You're not passing `x` into `__new__`, and it's unlikely that Python specifies what list is supposed to do if you give init and new different arguments.

It's much safer just to call the list constructor - there is no benefit to splitting up the call here, and it makes you vulnerable to implementation details in python itself.

Reply:
I see - I'd made the assumption that `list.__new__()` did nothing with its arguments, but I do see that this is an implementation detail.

Reply:
PR #7257 opened to make this change.